plugins/net_plugin/include/eosio/net_plugin/net_utils.hpp

```diff
@@ -40,29 +40,71 @@ namespace detail {
       return block_sync_rate_limit;
    }
 
+   /// @return host, port, remainder
+   inline std::tuple<std::string, std::string, std::string> split_host_port_remainder(const std::string& peer_add, bool should_throw) {
+      using std::string;
+      // host:port[:trx|:blk][:<rate>]
+      if (peer_add.empty()) return {};
+
+      auto colon_count = std::count(peer_add.begin(), peer_add.end(), ':');
+      string::size_type end_bracket = 0;
+      if (peer_add[0] == '[') {
+         end_bracket = peer_add.find(']');
+         if (end_bracket == string::npos) {
+            EOS_ASSERT(!should_throw, chain::plugin_config_exception,
+                       "Invalid address specification ${a}, IPv6 no closing square bracket", ("a", peer_add) );
+            return {};
+         }
+      } else if (colon_count >= 7) {
+         EOS_ASSERT(!should_throw, chain::plugin_config_exception,
+                    "Invalid address specification ${a}; IPv6 addresses must be enclosed in square brackets.", ("a", peer_add));
+         return {};
+
+      }
+      string::size_type colon = peer_add.find(':', end_bracket+1);
+      if (colon == string::npos || colon == 0) {
+         return {};
+      }
+      string::size_type colon2 = peer_add.find(':', colon + 1);
+      string host = (end_bracket > 0) ? peer_add.substr( 0, end_bracket+1 ) : peer_add.substr( 0, colon );
```

Comment:
This would successfully parse a peer spec such as:
`"[2001:db8:85a3:8d3:1319:8a2e:370:7348]whatever junk I want here:9876:250KB/s"`

Which could be fixed by checking that, `end_bracket == 0 || end_bracket+1 == colon`.

Then this line could be simpler (the only case where `end_bracket > 0 && end_bracket+1 != colon` if when there is junk text between the end bracket and the colon):

`string host = peer_add.substr( 0, colon );`

Reply:
https://github.com/AntelopeIO/spring/pull/1115/commits/6c2442b14f7724c44625cc8b8f862b1cada1449d